pkg/result/result_helper.go

```diff
@@ -121,18 +155,31 @@ func (r errorOut) GetError() error {
 	return r.err
 }
 
+// Continue indicates that the current step in the reconciliation is done. The caller should proceed with the next step.
 func Continue() ReconcileResult {
 	return continueReconcile{}
 }
 
+// Done indicates that the entire reconciliation loop was successful.
```

Comment:
Issue: from a design perspective, this seems tricky, since only the last step should ever call `Done()` based on this description. No earlier step would ever know that the entire reconcilation was `Done()` from what I can see? 

I think if we're getting into this topic, we should make `Done()` more descriptive - perhaps it needs a rename to `AllReconciliationDone()` or something similar to that, which indicates that all reconciliation steps to reconcile a given object (from the `req` passed to the controller) is done. 

We need to be explicit about WHAT is done - that's the issue we're currently struggling with. Naming these implementations correctly might be a good starting point.

Reply:
> No earlier step would ever know that the entire reconcilation was Done() from what I can see?

This is exactly what it is, the reconciling process is done. It's not even really a necessary process didn't end in a pkg, but in the Reconcile() function. Just return reconcile.Result{}, nil.

> We need to be explicit about WHAT is done - that's the issue we're currently struggling with. Naming these implementations correctly might be a good starting point.

Are we having issues? This API didn't even begin here and the original usage had no issues. I'm not convinced we need this ReconcileResult API at all, but then that's not what we're discussing here.

Reply:
> Are we having issues? This API didn't even begin here and the original usage had no issues. I'm not convinced we need this ReconcileResult API at all, but then that's not what we're discussing here.

Maybe this should be something we're discussing. We are definitely having issues because there is no consistency in the usage of this API in the codebase and there are regular disagreements on what Done, Continue, and Completed mean, as there have been for several years.

Reply:
> No, there hasn't been

At the very least there's been some ambiguity, as evidenced by the changes we are doing in this PR to align different parts of the code.

> it had a clear definition there how things worked

This might be your longer experience of cass-operator speaking, but personally I wouldn't call this API clear, it took way more effort to understand than it should.
Anyway, whether it is clear or not, I don't see any downside to documenting the intent.

Reply:
Let's focus on whether or not we agree with the comments that @olim7t added on each function.
@burmanm @Miles-Garnsey, do these comments look good? 

Let's not change names for now and just build a common understanding.